lib/rubocop/cop/sorbet/obsolete_strict_memoization.rb

```diff
@@ -0,0 +1,129 @@
+      #   sig { returns(Foo) }
+      #   def foo
+      #     @foo ||= T.let(Foo.new, T.nilable(Foo))
+      #   end
+      #
+      # TODO: disable this cop when the Sorbet version is older than `0.5.10210.20220725105033-6e80c502e`.
+      # TODO: find the right way to access the line length limit, indentation style, and indentation width.
+      class ObsoleteStrictMemoization < RuboCop::Cop::Base
+        include RuboCop::Cop::MatchRange
+
+        MESSAGE = "This two-stage workaround for memoization in `#typed: strict` files is no longer necessary. " \
+          "See https://github.com/sorbet/sorbet/issues/995."
+
+        # @!method legacy_memoization_pattern?(node)
+        def_node_matcher :legacy_memoization_pattern?, <<~PATTERN
+          (begin
+            (ivasgn $_ivar                                # @_ivar = ...
+              (send                                       # T.let(_ivar, T.nilable(_ivar_type))
+                (const nil? :T) :let
+                (ivar _ivar)
+                (send                                     # T.nilable(_ivar_type)
+                  (const nil? :T) :nilable $_ivar_type)))
+            (or-asgn                                      # @_ivar ||= _initialization_expr
+              (ivasgn _ivar)
+              $_initialization_expr))
+        PATTERN
+
+        def on_begin(node)
+          return unless legacy_memoization_pattern?(node)
+
+          add_offense(node, message: MESSAGE)
+        end
+
+        def autocorrect(node)
+          ->(corrector) {
+            expression = legacy_memoization_pattern?(node)
+            ivar, ivar_type, initialization_expr = expression
+
+            base_indent = infer_base_indentation(node)
+
+            is_multiline_init_expr = initialization_expr.line_count != 1
+
+            correction = if is_multiline_init_expr
+              render_multi_line_correction(node, ivar, ivar_type, initialization_expr, base_indent)
+            else
+              single_line_correction =
+                "#{ivar} ||= T.let(#{initialization_expr.source}, T.nilable(#{ivar_type.source}))"
+
+              if (base_indent.length + single_line_correction.length) <= max_line_length
+                single_line_correction
+              else # The single-line correction was too long. Re-render it as a multi-line correction.
+                render_multi_line_correction(node, ivar, ivar_type, initialization_expr, base_indent)
+              end
+            end
+
+            corrector.replace(node, correction)
+          }
+        end
+
+        private
+
+        def single_indent
+          @single_indent ||= begin
+            ident_style_config = config.for_cop("Layout/IndentationStyle")
+
+            case ident_style_config["EnforcedStyle"]
+            when "tabs"
+              "\t"
+
+            when "spaces", nil
+              space_width = ident_style_config["IndentationWidth"] ||
+                config.for_cop("Layout/IndentationWidth")["Width"] || 2
```

Comment:
This `||` 2 is a bit of a hack.

Is there an API for accessing the config of another cop, and having it automatically apply its own efault value (if there's no explicitly configured value)?

Barring that, is there a way to access a default value for another cop's config?